File: src/Module/NodeValue/NodeValueEntity.java
package Module.NodeValue;

import Module.Node.NodeEntity;
import Module.Node.NodeModel;

import java.io.Serializable;
import java.sql.Timestamp;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
public class NodeValueEntity implements Serializable {
    public int id;
    public Integer value;
    public Timestamp time;
    public Integer nodeId;
    public NodeEntity nodeEntity;

    public NodeValueEntity() {
    }

    public NodeValueEntity(int id, Integer value, Timestamp time, Integer nodeId) {
        this.id = id;
        this.value = value;
        this.time = time;
        this.nodeId = nodeId;
    }

    public NodeValueEntity(NodeValueModel NodeValueModel, Object... objects) {
        this.id = NodeValueModel.getId();
        this.value = NodeValueModel.getValue();
        this.time = NodeValueModel.getTime();
        for (Object object : objects) {
            if (object instanceof NodeModel) {
                this.nodeEntity = new NodeEntity((NodeModel) object);
            }
        }
    }

    public NodeValueModel toModel() {
        NodeValueModel NodeValueModel = new NodeValueModel();
        NodeValueModel.setId(id);
        NodeValueModel.setTime(time);
        NodeValueModel.setNodeId(nodeId);
        NodeValueModel.setValue(value);
        return NodeValueModel;
    }
}

File: web/Angular4/src/app/Modules/NodeValue/NodeValue.Entity.ts
import {NodeEntity} from "../Node/Node.Entity";

export class NodeValueEntity {
    id: number;
    value: string;
    time: string;
    nodeId: string;
    NodeEntity: NodeEntity;
    IsEdit: boolean;
    IsActive: boolean = false;
    IsSelected: boolean = false;

    constructor(NodeValue: any = null) {
        if (NodeValue == null) {
            this.id = null;
            this.value = null;
            this.time = null;
            this.nodeId = null;
            this.NodeEntity = null;
        } else {
            this.id = NodeValue.id;
            this.value = NodeValue.value;
            this.time = NodeValue.time;
            this.nodeId = NodeValue.nodeId;
            this.NodeEntity = NodeValue.NodeEntity;
            // if (this.NodeEntity == null) n;
        }
        this.IsEdit = false;
    }
}